### PR TITLE
feat(howto): add platform development howto

### PR DESCRIPTION
**Pull request by morphis**

### CODE REVIEW

File: howto/anbox/landing.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The guides in this section describe how to work with the Anbox runtime.
```
Comment (ru-fu) [inline howto/anbox/landing.md:1]: We should add an explanation of what we mean with "Anbox runtime".

File: howto/landing.md [hunk 2]
```diff
@@ -6,7 +6,8 @@ The how-to guides in this section give directions on how to achieve your goals w
 | [How to install the Anbox Cloud Appliance](https://discourse.ubuntu.com/t/how-to-install-the-anbox-cloud-appliance/29702)| Instructions for deploying and configuring the Anbox Cloud Appliance on a cloud platform|
 | [How to update your installation](https://discourse.ubuntu.com/t/update-your-installation/24331)| Instructions for updating Anbox Cloud or the Anbox Cloud Appliance|
 | [How to manage Anbox Cloud](https://discourse.ubuntu.com/t/manage-anbox-cloud/24337) | Instructions for managing and working with Anbox Cloud or the Anbox Cloud Appliance |
-| [How to manage applications](https://discourse.ubuntu.com/t/manage-applications/24333) | Instructions for  managing applications and the Anbox Application Registry (AAR) |
+| [How to manage applications](https://discourse.ubuntu.com/t/manage-applications/24333) | Instructions for managing applications and the Anbox Application Registry (AAR) |
+| [How to work with the Anbox runtime](TBD) | Instructions on working with the Anbox runtime |
```
Comment (ru-fu) [inline howto/landing.md:10]: Also need to add both pages to index.md

File: howto/anbox/develop-platform.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+Anbox Cloud provides a platform SDK which allows the development of custom platform plugins for the Anbox runtime.
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:1]:
```suggestion
Anbox Cloud provides a platform SDK that allows the development of custom platform plugins for the Anbox runtime.
```

We should also link to https://anbox-cloud.io/docs/ref/platforms and give a short explanation and maybe some examples on why you would want to add your own platform.

File: howto/anbox/develop-platform.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+This guide assumes all steps are being run on a machine running Ubuntu 22.04 and hosting the [Anbox Cloud Appliance](https://discourse.ubuntu.com/t/install-appliance/22681).
+
+A platform module must be build on the same version of Ubuntu as the Anbox runtime. This means if you're using one of the Ubuntun 22.04 based Anbox images (e.g. `jammy:android12:arm64`) you must build on Ubuntu 22.04.
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:5]:
```suggestion
A platform module must be built on the same version of Ubuntu as the Anbox runtime. This means if you're using one of the Anbox images based on Ubuntu 22.04 (for example, `jammy:android12:arm64`), you must build on Ubuntu 22.04.
```
I guess that means the Anbox runtime runs inside the Ubuntu container? Adding that might make it easier to understand; I had to read this three times to understand that we're talking about the images first and then about the appliance in the next sentence.

File: howto/anbox/develop-platform.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+In case that you're not running the [Anbox Cloud Appliance](https://discourse.ubuntu.com/t/install-appliance/22681) on an Ubuntu 22.04 system, you can build the platform on a separate system, e.g. in a LXD or docker container or another machine.
+
+To get started you first need to clone the [Anbox Platform SDK](https://github.com/anbox-cloud/anbox-platform-sdk) git repository:
+
+    git clone https://github.com/anbox-cloud/anbox-platform-sdk.git
+
+Before you can build one of the examples, you have to install all necessary build dependencies:
+
+    sudo apt install -y --no-install-recommends cmake build-essential ninja-build dh-exec cmake-extras libgmock-dev libelf-dev
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:15]: Just link to https://anbox-cloud.io/docs/ref/sdks#anbox-platform-sdk for installation.
(Might need to update the dependencies there?)

File: howto/anbox/develop-platform.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+The [Anbox Platform SDK](https://github.com/anbox-cloud/anbox-platform-sdk) comes with various examples which demonstrate different features. The following steps are using the `minimal` example. Alternatively you can choose the `nvidia` example if you're working with NVIDIA GPUs and want to see graphical output.
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:17]:
```suggestion
The [Anbox Platform SDK](https://github.com/anbox-cloud/anbox-platform-sdk) comes with various examples that demonstrate different features. The following steps use the `minimal` example. Alternatively, choose the `nvidia` example if you're working with NVIDIA GPUs and want to see graphical output.
```

File: howto/anbox/develop-platform.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        -G Ninja \
+        -B build \
+        -DCMAKE_PREFIX_PATH=../../lib/anbox-platform-sdk/ \
+        -DCMAKE_INSTALL_LIBDIR=lib/${triplet} \
+        -DCMAKE_INSTALL_PREFIX=/usr examples/minimal/
+    ninja -C build
+
+As a result of the build process you will find a `platform_minimal.so` module in the `build` directory.
+
+[AMS](https://discourse.ubuntu.com/t/about-ams/24321) allows launching Anbox containers in a special development mode which is meant for development tasks, specifically for addon or platform development. The Anbox runtime is instructed, when in development mode, not to terminate the container when it detects failures or other problems.
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:32]: Dev mode is documented here: https://anbox-cloud.io/docs/exp/containers#dev-mode

File: howto/anbox/develop-platform.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+To try out the minimal platform start a raw container with development mode turned on:
+
+    amc launch --raw --devmode --instance-type=a4.3
+
+If you choose the `nvidia` example you need to also selected an [instance type](https://discourse.ubuntu.com/t/instance-types/17764) which supports GPUs:
+
+    amc launch --raw --devmode --instance-type=g4.3
+
+As result, the command will print out the id of the container. Please take a note of the id as you will need it in the next step
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:42]:
```suggestion
To try out the minimal platform, start a raw container with development mode turned on:

    amc launch --raw --devmode --instance-type=a4.3

If you choose the `nvidia` example, you must select an [instance type](https://discourse.ubuntu.com/t/instance-types/17764) that supports GPUs:

    amc launch --raw --devmode --instance-type=g4.3

The command will print out the ID of the container. Note down this ID; you will need it in the next step.
```

File: howto/anbox/develop-platform.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    [ 2022-12-07 14:00:18] [anbox-session] [base_platform.cpp:34@create] Using platform 'minimal'
+
+This shows that the platform was successfully loaded and is in use by Anbox.
+
+When doing development of a new platform you can also directly invoke the Anbox runtime rather than starting it via systemd. For that you first have to stop the `anbox.service` unit
+
+    amc exec "${id}" -- systemctl stop anbox
+
+Now you can start the Anbox runtime directl by running
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:80]:
```suggestion
When developing a new platform, you can also directly invoke the Anbox runtime rather than start it via `systemd`. For that, stop the `anbox.service` unit first:

    amc exec "${id}" -- systemctl stop anbox

Then start the Anbox runtime directly by running:
```

File: howto/anbox/develop-platform.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+    git clone https://github.com/anbox-cloud/anbox-platform-sdk.git
+
+Before you can build one of the examples, you have to install all necessary build dependencies:
+
+    sudo apt install -y --no-install-recommends cmake build-essential ninja-build dh-exec cmake-extras libgmock-dev libelf-dev
+
+The [Anbox Platform SDK](https://github.com/anbox-cloud/anbox-platform-sdk) comes with various examples which demonstrate different features. The following steps are using the `minimal` example. Alternatively you can choose the `nvidia` example if you're working with NVIDIA GPUs and want to see graphical output.
+
+To build the `minimal` platform example, run the following commands:
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:19]: Should we point out to users that they should do some changes to the code first, or that this is the place where they would do their code changes?

File: howto/anbox/develop-platform.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
+Anbox Cloud provides a platform SDK which allows the development of custom platform plugins for the Anbox runtime.
+
+This guide assumes all steps are being run on a machine running Ubuntu 22.04 and hosting the [Anbox Cloud Appliance](https://discourse.ubuntu.com/t/install-appliance/22681).
+
+A platform module must be build on the same version of Ubuntu as the Anbox runtime. This means if you're using one of the Ubuntun 22.04 based Anbox images (e.g. `jammy:android12:arm64`) you must build on Ubuntu 22.04.
+
+In case that you're not running the [Anbox Cloud Appliance](https://discourse.ubuntu.com/t/install-appliance/22681) on an Ubuntu 22.04 system, you can build the platform on a separate system, e.g. in a LXD or docker container or another machine.
+
+To get started you first need to clone the [Anbox Platform SDK](https://github.com/anbox-cloud/anbox-platform-sdk) git repository:
```
Comment (ru-fu) [inline howto/anbox/develop-platform.md:9]: Currently, the whole page is one chunk of text, which is very hard to follow - so we'll need to break this up.
I think it should work to use steps here (So "1. Install, 2. Build the example, 3. Launch" and so on, each with a few paragraphs and commands under them). If each step gets too long, we could use sections instead, but I think a list of steps should be fine for now.

**Review by ru-fu — COMMENTED** [review]
I did a quick first pass - overall, I think the placement should work (if we expect more how-tos for the Anbox runtime in the future?), but we need to be a bit more detailed about what we actually want to do here.

File: howto/anbox/develop-platform.md (hunk 5 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Done

File: howto/anbox/develop-platform.md (hunk 10 above)
Comment (morphis) [reply howto/anbox/develop-platform.md:17]: That's not needed to run the example as it's intended to run as-is.

File: howto/anbox/develop-platform.md (hunk 7 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Added

File: howto/anbox/landing.md (hunk 1 above)
Comment (morphis) [reply howto/anbox/landing.md]: Added

File: howto/anbox/develop-platform.md (hunk 7 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Added

File: howto/anbox/develop-platform.md (hunk 10 above)
Comment (morphis) [reply howto/anbox/develop-platform.md:17]: The example is meant to work as-is without any further changes.

File: howto/anbox/develop-platform.md (hunk 5 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Added

File: howto/anbox/develop-platform.md (hunk 11 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Added

File: howto/anbox/develop-platform.md (hunk 3 above)
Comment (morphis) [reply howto/anbox/develop-platform.md]: Added

File: howto/landing.md (hunk 2 above)
Comment (morphis) [reply howto/landing.md:10]: Ah yeah, added

**Comment by ru-fu** [comment]
I added a commit with some language and consistency changes - please have a quick look to check if you're happy with those changes. :)

**Comment by morphis** [comment]
LGTM, thanks @ru-fu

**Review by ru-fu — APPROVED** [review]
